File: store.py
import uuid
from datetime import datetime
import bcrypt
import config

# Base de datos en memoria (en producción usarías PostgreSQL, MongoDB, etc.)
users = {}           # id -> { id, email, password, role, created_at }
sessions = {}        # session_id -> { user_id, email, role, csrf_token, created_at, expires_at }
failed_attempts = {} # email -> { count, last_attempt, locked_until }

# Usuario admin precargado para demostración
_admin_id = str(uuid.uuid4())
users[_admin_id] = {
    "id": _admin_id,
    "email": "admin@passport.com",
    "password": bcrypt.hashpw(b"Admin1234!", bcrypt.gensalt(config.BCRYPT_ROUNDS)),
    "role": "admin",
    "created_at": datetime.now().isoformat(),
}

File: src/store.js
const bcrypt = require('bcryptjs');
const { v4: uuidv4 } = require('uuid');
const { BCRYPT_ROUNDS } = require('./config');

// Base de datos en memoria (en producción usarías PostgreSQL, MongoDB, etc.)
const users = new Map();          // id -> { id, email, password, role, createdAt }
const sessions = new Map();       // sessionId -> { userId, email, role, csrfToken, createdAt, expiresAt }
const failedAttempts = new Map(); // email -> { count, lastAttempt, lockedUntil }

// Usuario admin precargado para demostración
const adminId = uuidv4();
users.set(adminId, {
  id: adminId,
  email: 'admin@passport.com',
  password: bcrypt.hashSync('Admin1234!', BCRYPT_ROUNDS),
  role: 'admin',
  createdAt: new Date(),
});

module.exports = { users, sessions, failedAttempts };
